### DISCUSSION TITLE
Download link error

### DISCUSSION THREAD

**Discussion by kinzinho666** [filamentphp/filament#12409] [Help] [answered]
### Package

Table builder

### Package Version

v3.2

### How can we help you?

I use "RelationManager" and in this relationship I upload files, I would like to create a download link for these files, but I am getting an error as per images.

ps. file exists on dir, filesystem local

My code:
`<?php

namespace App\Filament\Resources\CompeticoeResource\RelationManagers;

use App\Models\Competicoe;
use App\Models\DocumentoComp;
use Filament\Tables\Actions\Action;
use Filament\Forms;
use Filament\Forms\Components\Hidden;
use Filament\Forms\Form;
use Filament\Resources\RelationManagers\RelationManager;
use Filament\Tables;
use Filament\Tables\Table;
use Illuminate\Contracts\Cache\Store;
use Illuminate\Contracts\Support\Htmlable;
use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\SoftDeletingScope;
use Illuminate\Support\Facades\Storage;

class DocumentoCompsRelationManager extends RelationManager
{
    protected static string $relationship = 'documento_comps';
    //Menu
    protected static ?string $navigationLabel = 'Documentos';
    //Heading page
    protected static ?string $modelLabel = 'Documento';
    protected static ?string $pluralModelLabel = 'Documentos';

    protected static ?string $icon = 'heroicon-o-document';

    protected static ?string $title = 'Documentos';
    protected static ?string $tableHeading = 'Documentos';

    public function form(Form $form): Form
    {
        return $form
            ->schema([
                Hidden::make('competicoe_id')
                ->default(Competicoe::find(request()->route('id'))?->id),
                Forms\Components\TextInput::make('descricao')
                    ->label('Descrição')
                    ->columnSpanFull()
                    ->maxLength(255),
                Forms\Components\Section::make('Documento')
                    ->hiddenLabel()
                    ->collapsible()
                    ->schema([
                        Forms\Components\FileUpload::make('arquivo')
                        ->hiddenLabel()
                        ->minSize(20)
                        ->maxSize(6000)
                        ->hint('Tamanho máximo permitido: 5mb.')
                        ->previewable(false)
                        ->directory('competicoes/documentos')
                        ->downloadable(),
                    ]),
            ]);
    }

    public function table(Table $table): Table
    {
        return $table
            ->recordTitleAttribute('arquivo')
            ->columns([
                Tables\Columns\TextColumn::make('descricao'),
            ])
            ->filters([
                //
            ])
            ->headerActions([
                Tables\Actions\CreateAction::make(),
            ])
            ->actions([
                Action::make('download')
                    ->label('Download')
                    ->action(function ($record) {
                        return Storage::download($record->arquivo);
                    }),
                Tables\Actions\EditAction::make(),
                Tables\Actions\DeleteAction::make(),
            ])
            ->bulkActions([
                Tables\Actions\BulkActionGroup::make([
                    Tables\Actions\DeleteBulkAction::make(),
                ]),
            ]);
    }
}`

![download_filament_1](https://github.com/filamentphp/filament/assets/29714101/97eda7ae-c2b0-4bd9-a07d-aa7405c71a81)
![download_filament_2](https://github.com/filamentphp/filament/assets/29714101/e8de4e8f-209e-4663-ba4f-67e37ef74a98)

**Comment by leandrocfe** [comment]
try `Storage::url($record->arquivo)`

**Reply by kinzinho666** [reply]
Returned 200 but nothing happened, the file was not downloaded
![download_filament_3](https://github.com/filamentphp/filament/assets/29714101/e2f52edc-6536-4420-9ffe-3f862845d25d)

**Reply by leandrocfe** [reply]
`Storage::download('/storage/' . $record->arquivo);`
?

**Reply by kinzinho666** [reply]
No
Unable to retrieve the file_size for file at location: storage/competicoes/documentos/01HVVMPS1D07Z1X72BCJGF3X55.pdf.

**Reply by leandrocfe** [reply]
Did you run `php artisan storage:link`?
Is the `APP_URL` configured  in .env file?

**Reply by kinzinho666** [reply]
Yes i do both

**Reply by leandrocfe** [reply]
Are you able to download this file in the download link, in the Form?

**Reply by leandrocfe** [reply]
`return Response::download('storage/' . $record->arquivo);`

**Reply by kinzinho666** [reply]
use Illuminate\Http\Response;
return Response::download('storage/' . $record->arquivo);
Method Illuminate\Http\Response::download does not exist.

ill create a front page to test create a download link manually

**Reply by leandrocfe** [reply, marked as answer]
`use Illuminate\Support\Facades\Response`, actually

**Reply by kinzinho666** [reply]
Thanks @leandrocfe , now is working!
